src/lib/compression/zstd_decompressor.h

```diff
@@ -18,18 +18,15 @@ class zstd_decompressor : public STREAM
 	using STREAM::STREAM; // pull in constructors
 
 public:
-	bool write(const char* data, size_t len)
+	bool write_once()
```

Comment:
Basically this is just a loop unrolling. As a result, we need to hold on to `_inBuff` for longer.

This is potentially a bad idea, since we don't own that memory. I don't love it...